### DISCUSSION TITLE
Minimum documents needed to run bertopic.

### DISCUSSION THREAD

**Discussion by akarshghale** [MaartenGr/BERTopic#1517] [Q&A]
Hi,

I have noticed that bertopic throws errors when the number of documents is very low ex 3-5. How many documents ideally there should be for bertopic to run without any errors?

**Comment by MaartenGr** [comment]
That depends on your specific code and the settings of your hyperparameters. Generally, 3-5 is not really sufficient for a topic modeling task when it takes significantly less time to just label those 3-5 examples yourself. You could, however, use something like k-Means to cluster them which generally works better with smaller data. 

Having said that, what is your use case for topic modeling on 3-5 documents when labeling the documents yourself could take just a few minutes?

**Comment by akarshghale** [comment]
Actually, plan is to automate it so for some labels that has less documents
it shouldn't end up throwing an error. As of now just added a simple check
to skip topic modelling if it's lower than 15.
